--- switch_types.h	2024-10-12 16:31:07
+++ switch_types.h.new	2024-10-12 16:33:25
@@ -1907,7 +1907,8 @@
 	SMBF_READ_VIDEO_PATCH = (1 << 24),
 	SMBF_READ_TEXT_STREAM = (1 << 25),
 	SMBF_FIRST = (1 << 26),
-	SMBF_PAUSE = (1 << 27)
+	SMBF_PAUSE = (1 << 27),
+	SMBF_LAST = (1 << 28)  
 } switch_media_bug_flag_enum_t;
 typedef uint32_t switch_media_bug_flag_t;
 
@@ -2902,3 +2903,4 @@
  * For VIM:
  * vim:set softtabstop=4 shiftwidth=4 tabstop=4 noet:
  */
+
